components/authz-service/server/v2/policy_refresher.go

```diff
@@ -152,14 +152,8 @@ func (refresher *policyRefresher) RefreshAsync() error {
 
 // updates OPA engine store with policy
 func (refresher *policyRefresher) updateEngineStore(ctx context.Context) error {
-	// We need to remove project filters from the request context
-	// otherwise they will be applied for store updates.
-	// This will fail on service start context, so only remove projects if ok.
-	md, ok := metadata.FromIncomingContext(ctx)
-	if ok {
-		delete(md, "projects")
-		ctx = metadata.NewOutgoingContext(ctx, md)
-	}
+	// Engine updates need unfiltered access to all data.
```

Comment:
This comment is a bit confusing to me... We're inside Authz-service, does it even matter if this is set or not? Might be missing something 💭

Reply:
We aren't using the `ctx` to filter projects within the `V2SetPolicies` call so we can remove all of this (not sure why we had it before 🤷‍♂ )

Reply:
It is needed, because that method proceeds to get policies, roles, and rules from the store--and those would be filtered without this bit.

Reply:
... but what's needed to avoid that is removing it from the incoming Metadata. Outgoing metadata is irrelevant here, since there's no grpc client calls involved in this code path, or is there?